### PR TITLE
Fix for "Incorrect match from camera database #794"

### PR DESCRIPTION
**Pull request by john-davies**
Summary of changes:

1. Fix bug so that the IXUS 70 camera is matched correctly
2. Add two extra test cases - one for IXUS 70 camera that showed the problem and one for a camera with no model number in the string.

### CODE REVIEW

File: src/openMVG/exif/sensor_width_database/datasheet.hpp [hunk 1]
```diff
@@ -51,13 +51,26 @@ struct Datasheet
...
     if (this_maker.compare(rhs_maker) == 0)
     {
-      // Search for digit substring and if there is a match
+      // Extract digit substring from database entry
+      std::vector<std::string> vec_db_model;
+      stl::split(this_model, ' ', vec_db_model);
+      std::string db_model_digit_substring = "";
+      for (const std::string & db_sub_model : vec_db_model)
+      {
+        if( std::find_if(db_sub_model.begin(), db_sub_model.end(), isdigit) != db_sub_model.end() )
```
Comment (pmoulon) [inline src/openMVG/exif/sensor_width_database/datasheet.hpp]: ```if (```

File: src/openMVG/exif/sensor_width_database/datasheet.hpp [hunk 2]
```diff
@@ -51,13 +51,26 @@ struct Datasheet
...
+          break;
+        }
+      }
+
+      // Search for digit substring in image camera model and if there is a match
 
       const bool has_digit =
         std::find_if(rhs_model.begin(), rhs_model.end(), isdigit) != rhs_model.end();
-      if (!has_digit)
+      if (!has_digit || ( db_model_digit_substring == "" ) )
```
Comment (pmoulon) [inline src/openMVG/exif/sensor_width_database/datasheet.hpp]: ```(!has_digit || db_model_digit_substring.empty() )```

**Comment by john-davies** [comment]
Yes, those are fine thanks. I've updated the branch with the changes.

**Comment by pmoulon** [comment]
Thank you @john-davies 
Merged by https://github.com/openMVG/openMVG/commit/9b2b0c525e1a3716df56e539a1a47961abcc2931

**Pull request closed without merging**